hartshorn-core/src/main/java/org/dockbox/hartshorn/core/domain/Exceptional.java

```diff
@@ -201,7 +209,8 @@ private Exceptional(final Throwable throwable) {
      * @throws NullPointerException
      *         if value is not present and {@code other} is null
      */
-    public T get(final Supplier<? extends T> other) {
+    @NotNull
+    public T get(@NotNull final Supplier<? extends T> other) {
```

Comment:
To ensure the result of the supplier is also checked as non-null, you could change this to `@NonNull Supplier<@NonNull ? extends T> other`

Reply:
Didn't even know you could do this, will do!